databases/week3/meal.sql

```diff
@@ -0,0 +1,146 @@
+VALUES 
+(1, 2, 41, '2024-07-25', '555-123-4567', 'John Doe', 'johndoe@example.com'),
+(2, 4, 42, '2024-07-26', '555-234-5678', 'Jane Smith', 'janesmith@example.com'),
+(3, 3, 43, '2024-07-27', '555-345-6789', 'Alice Johnson', 'alicejohnson@example.com'),
+(4, 5, 44, '2024-07-28', '555-456-7890', 'Bob Brown', 'bobbrown@example.com'),
+(5, 2, 45, '2024-07-29', '555-567-8901', 'Carol White', 'carolwhite@example.com'),
+(6, 2, 41, '2024-07-30', '555-678-9012', 'Dave Black', 'daveblack@example.com'),
+(7, 1, 42, '2024-07-31', '555-789-0123', 'Eve Green', 'evegreen@example.com'),
+(8, 4, 43, '2024-08-01', '555-890-1234', 'Frank Blue', 'frankblue@example.com'),
+(9, 3, 44, '2024-08-02', '555-901-2345', 'Grace Yellow', 'graceyellow@example.com'),
+(10, 2, 45, '2024-08-03', '555-012-3456', 'Hank Pink', 'hankpink@example.com');
+
+--Get all meals
+SELECT * from meal;
+
+--Add a new meal
+INSERT INTO meal (title, description, location, `when`, max_reservations, price, created_date)
+VALUES 
+('French Gourmet Evening', 'An elegant evening of fine French cuisine including coq au vin, 
+ratatouille, and crème brûlée.', '987 Bistro Blvd, Paris, France', '2024-08-25 19:00:00', 25, 60.00, '2024-07-23');
+
+--Get a meal with any id, fx 1
+SELECT title FROM meal WHERE id=44;
+--update a meal with any id, fx 1. Update any attribute fx the title
+-- or multiple attributes
+UPDATE meal set title='ghormeSabzi' WHERE id=43;
+--Delete a meal with any id, fx 1
+DELETE FROM meal WHERE id=43;
+
+--Get all reservations
+SELECT * from reservation;
+
+--Add a new reservation
+INSERT INTO reservation (id, number_of_guests, meal_id, created_date, contact_phonenumber, contact_name, contact_email)
+VALUES 
+(11, 3, 45, '2024-08-01', '555-678-1234', 'Lucy Brown', 'lucybrown@example.com');
+
+--Get a reservation with any id, fx 1
+SELECT * FROM reservation WHERE id=1;
+--Update a reservation with any id, fx 1. Update any attribute fx the title or multiple 
+UPDATE reservation SET contact_name='Reza' WHERE id=1;
+
+--Delete a reservation with any id, fx 1
+DELETE FROM reservation where id=2;
+--Get all reviews
+SELECT * from Review;
+
+INSERT INTO Review (title, description, meal_id, stars, created_date)
+VALUES 
+('Exquisite French Gourmet', 'The French Gourmet Evening was an unforgettable experience with exquisite dishes and impeccable service.', 4
+6, 5, '2024-08-26');
+--Get a review with any id, fx 1
+SELECT * from Review WHERE id=50;
+
+--Update a review with any id, fx 1. Update any attribute fx the title or multiple attributes
+UPDATE Review set stars=5 WHERE id=50;
+
+--Delete a review with any id, fx 1
+DELETE FROM Review where id=50;
+
+--Additional queries
+--Get meals that has a price smaller than a specific price fx 90
+SELECT * FROM meal where price>25
+
+--Get meals that still has available reservations
+SELECT title from meal JOIN reservation on meal_id=meal.id where max_reservations>number_of_guests;
```

Comment:
In this part you need to compare max_reservations with the total amount of guests made all reservations for that meal. Remember that a meal can be shared by several groups of people.